ssc/Pos/Ssc/Toss/Logic.hs

```diff
@@ -70,7 +69,13 @@ verifyAndApplySscPayload pm eoh payload = do
             let slot = epochOrSlot (const 0) (indexToCount . siSlot) eos
             when (slotSecurityParam <= slot && slot < 2 * slotSecurityParam) $
                 resetShares
-    mapM_ putCertificate blockCerts
+
+    case payload of
+        (CommitmentsPayload _ payloadCerts) -> mapM_ putCertificate payloadCerts
+        (OpeningsPayload _ payloadCerts) -> mapM_ putCertificate payloadCerts
+        (SharesPayload _ payloadCerts) -> mapM_ putCertificate payloadCerts
+        (CertificatesPayload payloadCerts) -> mapM_ putCertificate payloadCerts
```

Comment:
Really minor wibble, but the parents around the constructors are not needed there.